dist/systemd/barrierc@.service.in

```diff
@@ -0,0 +1,59 @@
+#  Barrier Client service barrierc-@BARRIER_VERSION@
+#
+#  This file is part of Barrier.
+#
+#  This is a systemd template service for the barrierc client service. Instances
+#  are started using @<hostname>. For example:
+#
+#  Starting:
+#   systemctl start barrierc@myserver
+#  or:
+#   systemctl start barrierc@myserver:24800
+#
+#  Enabling:
+#   systemctl enable barrierc@myserver
+#
+#  SSL Fingerprints are stored in /var/lib/barrier@<hostname> for each client 
+#  instance.
+
+[Unit]
+Description=Barrier Client connected to %I (Open-source KVM software)
+Documentation=man:barrierc(1) man:barriers(1)
+Documentation=https://github.com/debauchee/barrier/wiki
+# Require network before starting barrierc
+After=network-online.target
+Wants=network-online.target
+# Don't run client if server is running
+Conflicts=barriers.service
+
+[Service]
+Type=exec
+# Log level may be FATAL, ERROR, WARNING, NOTE, INFO, DEBUG, DEBUG1, DEBUG2
+Environment=LOG_LEVEL=INFO
+# Default display is :0
+Environment=DISPLAY=:0
+# Store fingerprints in instance specific directories
+Environment=XDG_DATA_HOME=/var/lib/barrier/barrier@%i
+# TrustedServers.txt Directory
+Environment=FP_DIR=/var/lib/barrier/barrier@%i/barrier/SSL/Fingerprints
+# Ensure the Fingerprints directory exists
+ExecStartPre=mkdir -p "${FP_DIR}"
+
+# This uses openssl commands and grep to get the server's key and 
+# store it in the TrustedServers.txt file. OpenSSL is a requirement
+# for barrier on Linux so these commands should exist. This will only
+# work if using the default 24800 port (since the port number must be
+# specified for openssl)
+ExecStartPre=sh -c "[ -f "${FP_DIR}/TrustedServers.txt" ] ||\
+openssl s_client -connect %i:24800 2>/dev/null |\
+openssl x509 -noout -sha1 -fingerprint |\
+grep -oE '([A-Z0-9]{2}:?){20}' > ${FP_DIR}/TrustedServers.txt"
+
+# Main executable
+ExecStart=/usr/bin/barrierc --enable-crypto --display ${DISPLAY} --debug ${LOG_LEVEL} --no-daemon %i
```

Comment:
Note that if Barrier is installed via Snap, the path to the executable is `/snap/bin/barrier.barrierc` etc. Are there CMake tokens for this you could use?

Reply:
I'm not seeing any thing in the CMakeLists.txt, but I'm doing some reading on Snap to see if there's an environment variable or something that can be used to change the path with CMake.

Reply:
Snaps have native support for daemon services. Would that help?
https://snapcraft.io/docs/services-and-daemons